### ISSUE TITLE
error handling protobuf option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema)

### ISSUE THREAD

**Issue by baganokodo2022** [confluentinc/confluent-kafka-go#946]
Description
===========
confluent-kafka-go version v2.0.2
We make use of protobuf.Serializer in confluent-kafka-go to register proto schemas on Confluent Schema Registry. The serializer breaks on the option, `option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema)`, which should be rendered as,
`
  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
    json_schema: {
      title: "SimpleMessage"
      description: "A simple message."
      required: ["id"]
    }
  };
`
instead, it ended up with,
`
  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
    json_schema: <
      required: "order_id"
      required: "portfolio_id"
      required: "product_id"
      required: "type"
      required: "side"
      required: "created_time"
      required: "status"
    >
  };
`
CSR returned the following error message,
details: Could not parse Protobuf - Syntax error in :217:18: expected a word
line 217 points to `json_schema: <`

How to reproduce
================
Create a simple protobuf message with the above message-level option like this,
syntax = "proto3";

package org.order.common;

import "google/protobuf/descriptor.proto";
import "protoc-gen-swagger/options/openapiv2.proto";

message Order {
  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
    json_schema: {
      title: "SimpleMessage"
      description: "A simple message."
      required: ["id"]
    }
  };

  string id = 1; 
}

In golang,
client, err := schemaregistry.NewClient(schemaregistry.NewConfig(srUrl))
ser, err := protobuf.NewSerializer(client, serde.ValueSerde, protobuf.NewSerializerConfig())
payload, err := serializer.ser.Serialize(serializer.topic, <protoc_generated_Order_value>)

Checklist
=========
Please provide the following information:

 - [ ] confluent-kafka-go and librdkafka version (`LibraryVersion()`): v2.0.2
 - [ ] Apache Kafka broker version: 2.8.1
 - [ ] Client configuration: `ConfigMap{...}`
 - [ ] Operating system: Unbuntu 18
 - [ ] Provide client logs (with `"debug": ".."` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by baganokodo2022** [comment]
Today, I successfully reproduce the same error in a different code path,

![Screenshot 2023-02-16 at 2 43 35 PM](https://user-images.githubusercontent.com/116464387/219504058-b8ef3460-4650-4db9-bf8c-bdf2a4fa5f55.png)

**Comment by baganokodo2022** [comment]
The problem seems to be on this line,

https://github.com/golang/protobuf/blob/master/proto/text_encode.go#L373

If we replace it with,
```var bra, ket byte = '{', '}'```

It generated the desired schema,
```
syntax = "proto3";

package org.order.common;

import "google/protobuf/descriptor.proto";
import "protoc-gen-swagger/options/openapiv2.proto";

message Order {
  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
    json_schema: {
      title: "SimpleMessage"
      description: "A simple message."
      required: ["id"]
    }
  };
  string id = 1; 
}
```

**Comment by baganokodo2022** [comment]
according to the responses from https://github.com/golang/protobuf/issues/1523, the root cause is clearly on `jhump/protoreflect`.

**Comment by baganokodo2022** [comment]
filed the issue to jhump, https://github.com/jhump/protoreflect/issues/549

**Comment by jhump** [comment]
> the root cause is clearly on jhump/protoreflect

@baganokodo2022, I don't agree. The output in your examples is perfectly valid Protobuf source. Message literals in custom option values are allowed to use either angle brackets or curly braces for nested messages/groups.

So I think the root cause is more likely to be whatever is processing/parsing the proto sources in Confluent's schema registry. It is likely not completely compliant/compatible with what protoc (the reference compiler) considers valid source.

**Comment by baganokodo2022** [comment]
This issue is proven fixed in https://github.com/jhump/protoreflect/releases/tag/v1.15.0-rc2

**Comment by jhump** [comment]
FWIW, I think there is still a bug in Confluent that needs fixing: it should _not_ be complaining about the use of angle brackets as that **is** allowed in protobuf sources.

The above mentioned release of `github.comjhump/protoreflect`simply _works around_ the bug, in that its `protoprint` package will no longer produce files that use the angle brackets.
